randovania/games/prime1/exporter/patch_data_factory.py

```diff
@@ -232,17 +232,66 @@ def create_data(self) -> dict:
 
             for area in world.areas:
                 world_data[world.name]["rooms"][area.name] = {"pickups":[]}
-                
-                pickup_indices = sorted(node.pickup_index for node in area.nodes if isinstance(node, PickupNode))
-                if pickup_indices:
-                    world_data[world.name]["rooms"][area.name] = {
-                        "pickups": [
-                            prime1_pickup_details_to_patcher(pickup_list[index.index],
-                                                             index.index in modal_hud_override,
-                                                             self.rng)
-                            for index in pickup_indices
-                        ],
-                    }
+
+                def node_len(a: PickupNode):
+                    return a.pickup_index
+
+                pickup_nodes = sorted((node for node in area.nodes if isinstance(node, PickupNode)), key=node_len)
+                for node in pickup_nodes:
+                    pickup_index = node.pickup_index.index
+                    pickup = prime1_pickup_details_to_patcher(pickup_list[pickup_index],
+                                                              pickup_index in modal_hud_override,
+                                                              self.rng)
+                    if "position_required" in node.extra.keys() and node.extra["position_required"]:
```

Comment:
`node.extra.get` handles the of the key not existing nicely (second argument which is None by default)